Topics/Tech_Stacks/WebSockets/Web-Sockets.md

```diff
@@ -0,0 +1,169 @@
+
+WebSockets are particularly adept at handling use cases where the server needs to send data to the client proactively, without the client first making a request. This capability has led to the widespread adoption of WebSockets in various applications that require real-time functionality, such as live notifications, instant messaging, and collaborative editing platforms.
+
+In the context of CSC301, understanding WebSockets is essential for building modern applications where real-time user interaction is a core requirement. With WebSockets, software engineers can create responsive and interactive user experiences that were once difficult to achieve with traditional web technologies.
+
+![cpVnative](http_vs_websocket.png)
+
+## WebSockets vs HTTP
+HTTP, a foundational technology of the web, is designed around a stateless request-response model. This means that for each piece of data exchanged between the client and server, a new connection is initiated, followed by a request from the client and a response from the server. While effective for many web applications, this approach can introduce latency and inefficiency, especially in scenarios that demand real-time updates.
+
+In contrast, WebSockets offer a more dynamic communication protocol. After completing an initial handshake, WebSockets establish a stateful, two-way communication channel. This persistent connection remains open, allowing for continuous data exchange without the need to establish new connections for each interaction. This fundamental difference in operation between WebSockets and HTTP brings several notable advantages in specific contexts:
+
+- **Reduced Latency:** By eliminating the need for repeated HTTP handshakes, WebSockets significantly decrease latency, making them ideal for time-sensitive applications.
+- **Real-time Data Flow:** The open communication channel enables real-time data exchange. This is particularly beneficial for applications requiring immediate updates, such as online gaming, live sports updates, and financial trading platforms.
+- **Efficient Resource Utilization:** Operating over a single, persistent connection, WebSockets make more efficient use of server and network resources compared to the HTTP model, which opens and closes connections for each request/response pair.
+
+## Advantages and Disadvantages
+
+### Advantages
+- **Real-time Communication:** The essence of WebSockets is the ability to foster live interactions. This real-time capability is a game-changer for applications that require immediate user feedback, such as chat platforms, live sports updates, and multiplayer online games. 
```

Comment:
Clarify the term "real-time communication" by providing examples of applications that benefit from this capability.